llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -1646,22 +1646,19 @@ bool llvm::sortPtrAccesses(ArrayRef<Value *> VL, Type *ElemTy,
 
     // Check if the pointer with the same offset is found.
     int64_t Offset = *Diff;
-    auto Res = Offsets.emplace(Offset, Cnt);
-    if (!Res.second)
+    auto [It, IsInserted] = Offsets.emplace(Offset, Cnt++);
```

Comment:
I think the more idiomatic way to handle `Cnt` would be to loop over `enumerate()`.

Reply:
I didn't know that STLExtras had an `llvm::enumerate()`. Otherwise, the std version is only available in C++23. Fixed now.